tests/appsec/test_automated_login_events.py

```diff
@@ -1348,3 +1218,703 @@ def validate_iden(meta):
+        self.r_hdr_success = weblog.post(
+            "/login?auth=local",
+            data=login_data(context, USER, PASSWORD),
+            headers=HEADERS,
+        )
+
+    @missing_feature(context.library < "dotnet@3.7.0")
+    def test_login_success_headers(self):
+        # Validate that all relevant headers are included on user login success on extended mode
+
+        def validate_login_success_headers(span):
+            if span.get("parent_id") not in (0, None):
+                return
+
+            for header in HEADERS:
+                assert f"http.request.headers.{header.lower()}" in span["meta"], f"Can't find {header} in span's meta"
+            return True
+
+        interfaces.library.validate_spans(self.r_hdr_success, validate_login_success_headers)
+
+    def setup_login_failure_headers(self):
+        self.r_hdr_failure = weblog.post(
+            "/login?auth=local",
+            data=login_data(context, INVALID_USER, PASSWORD),
+            headers=HEADERS,
+        )
+
+    @missing_feature(context.library < "dotnet@3.7.0")
+    def test_login_failure_headers(self):
+        # Validate that all relevant headers are included on user login failure on extended mode
+
+        def validate_login_failure_headers(span):
+            if span.get("parent_id") not in (0, None):
+                return
+
+            for header in HEADERS:
+                assert f"http.request.headers.{header.lower()}" in span["meta"], f"Can't find {header} in span's meta"
+            return True
+
+        interfaces.library.validate_spans(self.r_hdr_failure, validate_login_failure_headers)
+
+
+@rfc("https://docs.google.com/document/d/1RT38U6dTTcB-8muiYV4-aVDCsT_XrliyakjtAPyjUpw")
+@scenarios.appsec_auto_events_extended
+@features.user_monitoring
+@features.user_id_collection_modes
+class Test_V3_Login_Events_Anon:
+    """Test login success/failure use cases
+    As default mode is identification, this scenario will test anonymization.
+    """
+
+    def setup_login_success_local(self):
+        self.r_success = weblog.post("/login?auth=local", data=login_data(context, USER, PASSWORD))
+
+    def test_login_success_local(self):
+        assert self.r_success.status_code == 200
+        for _, trace, span in interfaces.library.get_spans(request=self.r_success):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.login.success.usr.login"] == USERNAME_HASH
+            assert meta["_dd.appsec.usr.login"] == USERNAME_HASH
+            assert meta["_dd.appsec.events.users.login.success.auto.mode"] == "anonymization"
+            assert meta["appsec.events.users.login.success.track"] == "true"
+
+            # optional (to review for each library)
+            if context.library not in libs_without_user_id:
+                assert meta["usr.id"] == "social-security-id"
+                assert meta["_dd.appsec.usr.id"] == "social-security-id"
```

Comment:
this is the anon class no ? you're even checking that the collection mode is anonymization just a few lines above
```suggestion
                assert meta["usr.id"] == "anon_5f31ffaf95946d2dc703ddc96a100de5"
                assert meta["_dd.appsec.usr.id"] == "anon_5f31ffaf95946d2dc703ddc96a100de5"
```